File: Hard/Java/2141. Maximum Running Time of N Computers.java
class Solution {
    public long maxRunTime(int n, int[] batteries) {
        Arrays.sort(batteries);
        long sum = 0;
        int k = 0;
        int len = batteries.length;

        for(int b : batteries)
            sum += b;

        while(sum / (n - k) < batteries[len - 1 - k]){
            sum -= batteries[len - 1 - k];
            k++;
        }
        
        return sum / (n - k);
    }
}

File: Hard/Python/2141. Maximum Running Time of N Computers.py
class Solution(object):
    def maxRunTime(self, n, batteries):
        batteries.sort()
        total = 0
        k = 0
        length = len(batteries)

        for b in batteries:
            total += b
        
        while total / (n - k) < batteries[length - 1 - k]:
            total -= batteries[length - 1 - k]
            k += 1

        return total / (n - k)
